src/sanctum/src/Http/Middleware/AuthenticateSession.php

```diff
@@ -33,14 +30,27 @@ public function __construct(
      */
     public function process(ServerRequestInterface $request, RequestHandlerInterface $handler): ResponseInterface
     {
-        if (! $this->session->isStarted() || ! $user = $request->getAttribute('user')) {
+        if (! $this->session->isStarted()) {
             return $handler->handle($request);
         }
 
         $guards = Collection::make(Arr::wrap(config('sanctum.guard')))
             ->mapWithKeys(fn ($guard) => [$guard => $this->auth->guard($guard)])
             ->filter(fn ($guard) => $guard instanceof SessionGuard);
 
+        // Get the authenticated user from the guards
+        $user = null;
```

Comment:
What will happen if we just get user from `AuthFactory` directly? This part is extra logic compared to Laravel's sanctum?

Reply:
You're right, I think the implementation in Laravel's version doesn't work in this scenario. This version should be better.